### PR TITLE
Futurization

### PR DESCRIPTION
**Pull request by pmlopes**
Upgrade API to support vert.x 4.0 Future

### CODE REVIEW

**Comment by vietj** [comment]
are there APIs that needs to be deprecated here ?

File: tools/redis-api.hbs [hunk 1]
```diff
@@ -37,13 +37,25 @@ public interface RedisAPI {
...
+  /**
+   * Redis command <a href="https://redis.io/commands/{{ name }}">{{ camelcase safename }}</a>.
+   * @return Future response.
+   */
+  default Future<@Nullable Response> {{ camelcase safename }}({{{ types }}}) {
+    return send(Command.{{ enum }}{{#if argLen}}, {{{ args }}}{{#if variable}}.toArray(){{/if}}{{/if}});
+  }
 {{/each}}
 
   @GenIgnore
```
Comment (vietj) [inline tools/redis-api.hbs]: should't we remove @GenIgnore ?

Comment (pmlopes) [reply tools/redis-api.hbs]: we can't this a `varargs` method which is handy to delegate all synthetic API calls as the compiler enforces the types

Comment (vietj) [reply tools/redis-api.hbs]: oh right, we need to support varargs at some point in codegen :-)

**Pull request merged**